scalableminds/webknossos
# Enable maximized gl components

This PR enables the maximize functionality of golden layout. The changes are kept to a minimum by only changing CSS properties. Except for the maximized item, the items must be hidden (this includes the header of a golden layout component). To perform this the content of the page is directly edited via the DOM interface.

### known issues
- The 3d viewport does not render correctly when maximized, as the other viewports are not rendered.
- After resuming from the maximized version the 3d Viewport is broken.

### URL of deployed dev instance (used for testing):
- [https://enablemaximizedglcomponents.webknossos.xyz](https://enablemaximizedglcomponents.webknossos.xyz)

### Steps to test:
- open any kind of tracing view.
- arrange the components of the view freely.
- maximize one component by using the maximize icon in on the right side of the header of a component.
- If a stack is maximized, changing tabs should work out and should not close the maximized view.
- There should not be any borders or headers of "hidden" components visible.
- leaving the maximized view should work out fine and the layout should be the same as before.
- The maximized state should be saved, when reloading (assuming the layout was saved via auto save or manually)

### Issues:
- fixes #3743

------
